src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -623,8 +624,10 @@ internal Boolean LoadLocalHeaderExtraFieldAndCompressedBytesIfNeeded()
                 _compressedBytes = new Byte[(_compressedSize / MaxSingleBufferSize) + 1][];
                 for (int i = 0; i < _compressedBytes.Length - 1; i++)
                 {
-                    _compressedBytes[i] = new Byte[MaxSingleBufferSize];
+                    _compressedBytes[i] = ArrayPool<byte>.Shared.Rent(MaxSingleBufferSize);
```

Comment:
Two comments

I don't think we should be making classes Disposable so that pooling is 100% effective in reclaiming memory.    Ideally only things that have SEMANTIC state change on death (e.g. files close themselves), should be Disposable.  I think it is OK to leak from the pool on Exceptions (since raising exceptions is supposed to be rare).   I believe this is Stephen's view as well. 

We really should have some mechanism where ArrayPool will release things from the pool if they have not been used (although it is hard to be perfect here, but we should probably release 'old' things from the pool that have not be used before a Gen2 GC (I believe we have an event).  

I also agree with Stephen that we should 'not make things worse'  in particular if we pool large arrays (e.g. > 1MB), we need some assurance that these will bel cleaned up if they are not used.